jobs/commons/src/utils/helpers.utils.ts

```diff
@@ -58,4 +58,24 @@ const bufferB64UrlEncode = (b: Buffer): string =>
   b.toString('base64')
     .replace(/=/g, "")
     .replace(/\+/g, "-")
-    .replace(/\//g, "_")
\ No newline at end of file
+    .replace(/\//g, "_")
+
+/**
+ * Zip two arrays based on a matching key
+ * Non-matching values are discarded.
+ * Note: the key value must be unique in array b
+ * @param a 
+ * @param b 
+ * @param getValueA Function that extracts the key for array a
+ * @param getValueB Function that extracts the key for array b
+ * @returns 
+ */
+export function zipBy<A, B, K>(a: A[], b: B[], getValueA: (a: A) => K, getValueB: (b: B) => K): [A, B][] {
+  const mapB = new Map<K, B>()
+
+  b.forEach(bv => mapB.set(getValueB(bv), bv))
+
+  return a
+    .map(av => [av, mapB.get(getValueA(av))])
+    .filter(([_, bv]) => bv !== undefined) as [A, B][] // TODO Is there a better way to match types?
```

Comment:
It's fine. There is no way to make Typescript automatically infer the result type of the `filter` expression. 

You could use the `is` keyword but it is kinda the same as using `as`.
Since here there is also an array decostruction in the parameters I would say that `as` is an accettable solution.